src/org/mozartoz/truffle/translator/Translator.java

```diff
@@ -181,6 +185,21 @@ public OzRootNode translateAST(String description, Statement ast, Function<OzNod
 	}
 
 	OzNode translate(StatOrExpr node) {
+		if(node instanceof TailMarkerStatement) {
+			//return translate(((TailMarkerStatement) node).call());
+
+			CallStatement call = ((TailMarkerStatement) node).call();
+			List<Expression> args = new ArrayList<>(toJava(call.args()));
+			OzNode[] argNodes = new OzNode[args.size()];
+			for (int i = 0; i < argNodes.length; i++) {
+				argNodes[i] = translate(args.get(i));
+			}
```

Comment:
A bit of duplicated logic here.
We should probably have a `ExecuteValuesNode` that just execute a OzNode[] and returns a Object[].